docs/docs/networking.md

```diff
@@ -94,13 +94,39 @@ See [the port definition section](#port-definition) for more information.
-Now let's say you have 3 instances of App ACME, for example 10.0.0.2:31000, 10.0.0.2:31001 and 10.0.0.3:31000.  There is a common need to have a port that will route with some algorithm to each of those instances.  This "port" is metadata which is specific to this application.  All instances of the application will be hosted behind this port.  In Marathon this is referred to as the `servicePort`.  As part of the ACME app definition you can specify a `servicePort` of :8080 for instance.  It is important to recognize that this is metadata and Marathon doesn't do anything with this information... which can be confusing.  It is however tracked and provided by Marathon.  The purpose is to provide metadata about the service, in this case the ACME app.  It is expected that a user will configure a load balancer to host a port :8080 (or the `servicePort`) and route to each of the instances of this application.  All the information is queryable from Marathon.  The marathon-lb service when configured, does exactly this.  marathon-lb will register all instances of an app and route to its configured `servicePort`.  In this configuration, a client will connect to a load balancer at port :8080 (`servicePort`) which will route (with some algorithm) to 10.0.0.2:31000 (`hostPort`) which will route to :8080 (`containerPort`) of the internal application.
+The following use cases explore how you can configure port mapping to meet your needs. 
 
-`VIP_0` defined in `portDefinitions` as a label is like `servicePort` in that it is informational and implementation dependent.  However, when used with DC/OS, services internal to DC/OS will make available to the cluster that DNS name and port as a routing to services.  An example configuration:
+You will configure 3 defined ports (`containerPort`, `hostPort`, and `servicePort`) and the label for `VIP_0` in your port mapping.
+
+All applications on Mesos are hosted in containers. Containers can be run in bridge or host mode.
+
+Let us assume an app called "ACME," and that the actual application opens port `:8080`.
+
+#### Host Mode Use Case
+
+When in host mode, the container will have the host network interface defined in the container namespace. The container will not do any port mapping and the application will open up port `:8080` on the host network interface. In this case, there is no container port and the `hostPort` is `:8080`.
+
+#### Bridge Mode Use Case
+
+If the container is run in bridge mode, it gets its own network interface and the `containerPort` is `:8080`. In bridge mode, the container will open a pseudo random port on the host and bridge/NAT communication to the container port.
+
+For example, lets say the host port port is `:31000`. Now, the `containerPort` is `:8080`, but the `hostPort` is `:31000`. Clients to this service will open port `:31000` on the host the service is running on. The container will route traffic to the internal port in the container, `:8080`.
```

Comment:
Why do we need `:` before the port number?

Reply:
@kensipe What was the rationale there?

Reply:
...removed them